pages/docs/style-guide.mdx

```diff
@@ -30,6 +32,109 @@ There is only one exception. Functions from module Function shall be used direct
 import { pipe } from "@effect/data/Function"
 ```
 
+## Writing Code with Effect
+
+Nested callbacks are a common problem in JavaScript, and are often handled by using async/await in most modern JavaScript codebases. However, since we are working in the world of Effects, the available APIs are a bit different.
+
+This section will cover the three common ways to organize Effectful code:
+
+1. Using the pipeable API
+2. Using Generators
+3. Using the `Do` simulation
+
+### 1. Using the pipeable API
+
+The pipe function is a common way to organize Effectful code. It is a function that takes a value and a list of functions, and applies the functions to the value in order.
+
+The pipe function is defined in the Effect library, and is available in the Effect namespace.
+
+```ts twoslash
+import { pipe } from "@effect/data/Function"
+import * as Effect from "@effect/io/Effect"
+import * as Random from "@effect/io/Random"
+
+const example = pipe(
+  Random.next(),
+  Effect.flatMap((n) =>
+    n > 0.5 ? Effect.succeed("yay!") : Effect.fail("oh no!")
+  )
+)
+```
+
+Here, Random.next() is getting piped as an argument to `Effect.flatMap`.
+
+<Callout>
+  Many functions in Effect have `dual` APIs, which means that you can either
+  call it `data-first` or `data-last`. Here, you can see how `Effect.flatMap`
+  was already defined, so `Random.next()` is passed in last. Similarly, you can
```

Comment:
```suggestion
  is _after_ the call to `Random.next()` in the `pipe`, so `Random.next()` is passed in last (i.e. `data-last`). Similarly, you can
```

Reply:
I was doing another read at my original copy and definitely wanted to change it, thanks for these changes! Will do another pass.